Cogburn/detections tweaks

coreyogburn opened the pull request:
Multiple Tweaks:
- Author Field Fix
- ClickToEdit UI Styling
- Add help link to Detections Tuning Page

coreyogburn committed:
Detection Author and ClickToEdit Class

When creating a custom detection, the author field will be filled in with the current user's First Name and Last Name and falling back to the user's Email. If a detection is a community rule, it'll list the ruleset as the author. If the detection is not a community rule, this author field is displayed.

Override fields now have the clicktoedit class that highlights and changes the cursor on mouseover. The class is removed while the element is being edited.

coreyogburn committed:
Override Help Button

Added a help button to open SO Documentation to Detection Overrides.

Noticed we imported a library we aren't using. Ran `go mod tidy`.

jertel commented on html/index.html:
```diff
@@ -1325,6 +1325,9 @@ <h3 class="text--primary">{{ i18n.commentAddDetection }}</h3>
                         </div>
                         <div v-if="newOverride" class="d-flex justify-end text-body-2 mt-2">
                           <div class="d-inline-flex">
+                            <a href="https://docs.securityonion.net/en/2.4/" target="_blank">
```
This doc link needs to use the local stored docs. See the grid UI for an example:
```
<v-btn icon :title="i18n.clickForMoreInformation" :href="$root.parameters.docsUrl + 'grid.html'" :target="$root.target('so-help')" >
```

jertel commented on html/js/i18n.js:
```diff
@@ -585,6 +585,7 @@ const i18n = {
       order: 'Order',
       osUptime: 'OS Uptime',
       other: 'Other',
+      overrideDocumentationHelp: 'Security Onion Documentation for Overrides',
```
The Grid UI uses this i18n message for it's help link:
```
clickForMoreInformation: 'Click for more information',
```
Where possible it would be better to use consistent language for help shortcuts. If you'd like to adjust the verbiage on this translation that is fine too.

coreyogburn committed:
Use `docsUrl` and Consistent Help Msg

Replaced hardcoded url's with docsUrl from the info handler.

Override Help message was updated for consistent text with the Grid page.
